flux_led/__main__.py

```diff
@@ -574,32 +576,68 @@ def _determineMode(self, ww_level, pattern_code):
                 mode = "color"
         elif pattern_code == 0x60:
             mode = "custom"
+        elif pattern_code == 0x41:
+            mode = "color"
         elif PresetPattern.valid(pattern_code):
             mode = "preset"
         elif BuiltInTimer.valid(pattern_code):
             mode = BuiltInTimer.valtostr(pattern_code)
 
         return mode
 
-    def update_state(self, retry=2):
+    def query_state(self, retry=2, led_type = None):
+
+        # default values
         msg = bytearray([0x81, 0x8a, 0x8b])
+        msg_len = 14
+
+        if self.protocol == 'LEDnet_wifi370' or led_type == 'LEDnet_wifi370':
+            msg =  bytearray([0xef, 0x01, 0x77])
+            msg_len = 11
+            led_type = 'LEDnet_wifi370'
+            self._use_csum = False
         try:
             self._send_msg(msg)
-            rx = self._read_msg(14)
+            rx = self._read_msg(msg_len)
```

Comment:
Ok, I was able to fix the code, but it does not play nice with tests.py. 

The test should allow for an initial response of 2 bytes, but that is not how mock handles there requests now. I will have to dig a little further.

quick question. This seems like optimization rather than bug squashing. does this need to be fixed before this pr can be merged? After all, this PR is getting quite large as it is.